server/services/auth_service.py

```
@@ -35,6 +36,14 @@ async def signup(self, signup_data: SignupRequest) -> SignupResponse:
             result = await self.auth_repository.create_user(signup_data)
         except InputValidationError:
             raise
+        except HTTPException as e:
+            # Re-raise HTTPException from repository with appropriate error type
+            if e.status_code == 400 and "already exists" in e.detail.lower():
+                raise ConflictError(e.detail)
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)